lib/Firtool/Firtool.cpp

```diff
@@ -131,9 +135,16 @@ LogicalResult firtool::populateCHIRRTLToLowFIRRTL(mlir::PassManager &pm,
 
   pm.nest<firrtl::CircuitOp>().addPass(firrtl::createPrefixModulesPass());
 
-  if (!opt.disableOptimization)
+  if (!opt.disableOptimization) {
     pm.nest<firrtl::CircuitOp>().addPass(firrtl::createIMConstPropPass());
 
+    pm.nest<firrtl::CircuitOp>().addPass(firrtl::createHoistPassthroughPass(
+        /*hoistHWDrivers=*/!opt.disableOptimization &&
+        !opt.disableHoistingHWPassthrough));
+    // Cleanup after hoisting passthroughs, for separation-of-concerns.
```

Comment:
I'd like to run this after the earlier use of this pass, but IMDCE can't be run that early in the pipeline presently.

Just to note:

This is run earlier to pave way for ensuring no input probes by the time we reach dedup (cc #6121).

Running after lower-types is where majority of non-HW signals are picked up presently, hence scheduling it here as well as part of "optimizing".

Might be good to run before IMCP actually, as this may provide new opportunities for constant-prop (different instances of a passthrough send different constants through, after hoisting we can see that and const-prop), but haven't explored that.

Reply:
I was thinking does it make sense to add a `firtool` test, for this second run of the pass. 
If its simple and small enough, otherwise not required.